src/db/collections/block.rs

```diff
@@ -247,35 +259,75 @@ impl BlockCollection {
+    pub num_blocks: u32,
+    /// The number of blocks referenced by a milestone that contain a payload.
+    pub num_tx_payload: u32,
+    /// The number of blocks containing a treasury transaction payload.
+    pub num_treasury_tx_payload: u32,
+    /// The number of blocks containing a milestone payload.
+    pub num_milestone_payload: u32,
+    /// The number of blocks containing a tagged data payload.
+    pub num_tagged_data_payload: u32,
+    /// The number of blocks referenced by a milestone that contain no payload.
+    pub num_no_payload: u32,
+    /// The number of blocks containing a confirmed transaction.
+    pub num_confirmed_tx: u32,
+    /// The number of blocks containing a conflicting transaction.
+    pub num_conflicting_tx: u32,
+    /// The number of blocks containing no transaction.
+    pub num_no_tx: u32,
 }
 
 impl BlockCollection {
-    /// Gathers block analytics.
-    pub async fn get_block_analytics<B: PayloadKind>(
-        &self,
-        start_index: Option<MilestoneIndex>,
-        end_index: Option<MilestoneIndex>,
-    ) -> Result<BlockAnalyticsResult, Error> {
-        let mut queries = vec![doc! {
-            "$nor": [
-                { "metadata.referenced_by_milestone_index": { "$lt": start_index } },
-                { "metadata.referenced_by_milestone_index": { "$gte": end_index } },
-            ],
-        }];
-        if let Some(kind) = B::kind() {
-            queries.push(doc! { "block.payload.kind": kind });
-        }
-        Ok(self
+    /// Gathers past-cone activity statistics for a given milestone.
+    pub async fn get_milestone_activity(&self, index: MilestoneIndex) -> Result<MilestoneActivityResult, Error> {
+        let mut activity: MilestoneActivityResult = self
             .aggregate(
-                vec![doc! { "$match": { "$and": queries } }, doc! { "$count": "count" }],
+                vec![
+                    doc! { "$match": { "metadata.referenced_by_milestone_index": index } },
+                    doc! { "$group": {
+                        "_id": null,
+                        "num_blocks": { "$count": {} },
+                        "num_tx_payload": { "$sum": {
+                            "$cond": [ { "$eq": [ "$block.payload.kind", "transaction" ] }, 1 , 0 ]
+                        } },
+                        "num_treasury_tx_payload": { "$sum": {
+                            "$cond": [ { "$eq": [ "$block.payload.kind", "treasury_transaction" ] }, 1 , 0 ]
+                        } },
+                        "num_milestone_payload": { "$sum": {
+                            "$cond": [ { "$eq": [ "$block.payload.kind", "milestone" ] }, 1 , 0 ]
+                        } },
+                        "num_tagged_data_payload": { "$sum": {
+                            "$cond": [ { "$eq": [ "$block.payload.kind", "tagged_data" ] }, 1 , 0 ]
+                        } },
+                        "num_no_payload": { "$sum": 0 },
```

Comment:
Reminder for myself

Reply:
what does that mean?

Reply:
The trick with "$not" that @Alexandcoats suggested works.

Reply:
Ah sorry, I didn't want to post that, it was in my pending review (notes).